arrow-libs/core/arrow-autoclose/src/commonTest/kotlin/arrow/AutoCloseTest.kt

```diff
@@ -3,62 +3,60 @@ package arrow
 import arrow.atomic.AtomicBoolean
 import io.kotest.assertions.throwables.shouldThrow
 import io.kotest.matchers.shouldBe
-import kotlinx.coroutines.CompletableDeferred
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.toList
 import kotlinx.coroutines.test.runTest
 import kotlin.coroutines.cancellation.CancellationException
 import kotlin.test.Test
 
-@OptIn(ExperimentalStdlibApi::class)
 class AutoCloseTest {
 
   @Test
   fun canInstallResource() = runTest {
-    val promise = CompletableDeferred<Throwable?>()
-    val wasActive = CompletableDeferred<Boolean>()
+    var throwable: Throwable? = RuntimeException("Dummy exception")
+    var wasActive = false
     val res = Resource()
 
     autoCloseScope {
       val r = autoClose({ res }) { r, e ->
-        promise.complete(e)
+        throwable = e
```

Comment:
Why is it OK to change here from `CompetableDeferred` to a regular `var`?

Reply:
The test doesn't `launch` any coroutines or anything, so the promise is unnecessary. This is in an effort to report failures faster (because if the promise.complete call was never made, the test just hangs until it times out, but now it fails fast)